server/punishments.ts

```diff
@@ -55,6 +55,19 @@ const GROUPCHAT_MONITOR_INTERVAL = 10 * 60 * 1000; // 10 minutes
  * A punishment is an array: [punishType, userid | #punishmenttype, expireTime, reason]
  */
 export type Punishment = [string, ID | PunishType, number, string];
+
+/**
+ * Info on punishment types. Can be used for either room punishment types or global punishments
+ * extended by plugins. It can either be just a string (corresponding to the display in /alts),
+ * or a [string, callback] - where the string is the /alts display, and the callback is what to do if the user
+ * _does_ have the punishment (can be used to add extra effects in lieu of something like a loginfilter.)
+ * Rooms will be specified for room punishments, otherwise it's null for global punishments
+ */
+export interface PunishInfo {
+	display: string;
```

Comment:
The main thing I don't like is `display`. I think `desc` would be clearer here.

Reply:
Desc it is. I renamed it.